chord_metadata_service/restapi/authz_middleware.py

```diff
@@ -31,12 +39,55 @@ def __call__(self, request):
                     }
                 }
             }
+            allowed_datasets = get_allowed_datasets(request_body)
+            if allowed_datasets is not None:
+                request.allowed_datasets = allowed_datasets
+        # auth for rego_dev_playground repo
+        elif (settings.CANDIG_OPA_VERSION == "rego_dev_playground" and
+              any(re.match(path_re, request.path) for path_re in AUTHZ_PATHS) and
+              request.method == "GET"):
+            tokens = {}
+            for header in request.headers:
+                header_all_caps = header.upper()
+                if header_all_caps.startswith(("X-CANDIG-LOCAL-", "X-CANDIG-DAC-", "X-CANDIG-FED-", "X-CANDIG-EXT-")):
+                    tokens[header] = json.loads(request.headers[header])
 
-            if settings.CANDIG_OPA_URL:
+            request_body = {
+                "input": {
+                    "headers": tokens,
+                    "body": {
+                        "path": request.path,
+                        "method": request.method
+                    }
+                }
+            }
+
+            allowed_datasets = get_allowed_datasets(request_body)
```

Comment:
So, it looks like this test is failing in Travis:
```FAIL: Tests the endpoint when OPA is down. Endpoint should return status code 500.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/CanDIG/katsu/.tox/python/lib/python3.6/site-packages/django/test/utils.py", line 373, in inner
    return func(*args, **kwargs)
  File "<string>", line 3, in wrapper
  File "/home/travis/build/CanDIG/katsu/chord_metadata_service/phenopackets/tests/test_phenopackets_api.py", line 141, in test_with_middleware_opa_is_down
    self.assertEqual(response.status_code, 500)
AssertionError: 200 != 500```

The issue seems to be that in function `get_allowed_datasets()`, when an exception occurs and that function returns, we're not checking if an error was raised here. We're treating it the same as if the function didn't raise an exception and returned allowed datasets.
If the function throws an exception, it should return the error response and we should return that error immediately here without executing any more code.